Frontend/ui-customer/src/Screens/CustomerUtils/RideSelectionScreen/View.purs

```diff
@@ -0,0 +1,261 @@
+loadButtonView state push =
+  linearLayout
+  [ height WRAP_CONTENT
+  , width MATCH_PARENT
+  , orientation VERTICAL
+  , background Color.white900
+  , onClick push (const Loader)
+  , clickable if state.data.loadMoreText == "LoadMore" then true else false
+  , gravity CENTER
+  , alignParentBottom "true,-1"
+  , padding (Padding 0 0 0 5)
+  , visibility if (state.props.loaderButtonVisibility && (not state.props.loadMoreDisabled)) then VISIBLE else GONE
+  ][ linearLayout[
+    background Color.grey900
+  , width MATCH_PARENT
+  , height $ V 1
+  ][]
+    ,linearLayout
+    [ height WRAP_CONTENT
+    , width WRAP_CONTENT
+    , orientation VERTICAL
+    , margin $ Margin 0 5 0 5
+    ]
+    [ textView $
+      [ width WRAP_CONTENT
+      , height WRAP_CONTENT
+      , text $ if state.data.loadMoreText == "LoadMore" then (getString LOAD_MORE) else (getString NO_MORE_RIDES)
+      , padding (Padding 10 5 10 5)
+      , color Color.blue900
+      ] <> FontStyle.body1 LanguageStyle
+    ]]
+
+ridesView :: forall w . PrestoList.ListItem -> (Action -> Effect Unit) -> ST.RideSelectionScreenState -> PrestoDOM (Effect Unit) w
+ridesView listItemm push state =
+  swipeRefreshLayout
+  ([height MATCH_PARENT
+  , width MATCH_PARENT
+  , onRefresh push (const Refresh)
+  ] <> if EHC.os == "IOS" then [] else [id "2000031"] )
+  [ Keyed.relativeLayout
+    [ width MATCH_PARENT
+    , height MATCH_PARENT
+    , orientation VERTICAL
+    ]([ Tuple "Rides"
+        $ PrestoList.list
+        [ height MATCH_PARENT
+        , scrollBarY false
+        , width MATCH_PARENT
+        , onScroll "rides" "RideSelectionScreen" push (Scroll)
+        , onScrollStateChange push (ScrollStateChanged)
+        , visibility $ case DA.null state.itemsRides of
+                    false -> VISIBLE
+                    true -> GONE
+        , PrestoList.listItem listItemm
+        , background Color.white900
+        , PrestoList.listDataV2 $ (DA.filter (\item -> (item.status) == toPropValue("COMPLETED") || (item.status) == toPropValue("CANCELLED")) state.prestoListArrayItems)
+        ]
+      , Tuple "NoRides"
+        $ linearLayout
+          [ height MATCH_PARENT
+          , width MATCH_PARENT
+          , background Color.white900
+          , visibility $ if (DA.length state.itemsRides) == 0  then VISIBLE else GONE
+          ][  ErrorModal.view (push <<< ErrorModalActionController) (errorModalConfig state)]
+      , Tuple "APIFailure"
+        $ linearLayout
+          [ height MATCH_PARENT
+          , width MATCH_PARENT
+          , background Color.white900
+          , gravity CENTER
+          , visibility if state.props.apiFailure then VISIBLE else GONE
+          ][  ErrorModal.view (push <<< APIFailureActionController) (apiErrorModalConfig state)]
+      , Tuple "LOADER"
+          $ PrestoAnim.animationSet
+          [ PrestoAnim.Animation
+            [ PrestoAnim.duration 1000
+            , PrestoAnim.toAlpha $
+                case state.shimmerLoader of
```

Comment:
Merge Identical cases, try 

`case state.shimmerLoader of
  loader | loader `elem` [ST.AnimatingIn, ST.AnimatedIn] -> 1.0
  loader | loader `elem` [ST.AnimatingOut, ST.AnimatedOut] -> 0.0`